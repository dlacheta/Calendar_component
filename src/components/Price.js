import React from "react";
import Rating from "./Rating";
import PropTypes from "prop-types";

const Price = ({ price, priceCountry, rating, ratingAmount }) => (
  <div className="price">
    <div>
      {price} {priceCountry} <span className="small_text">per night</span>
    </div>
    <Rating rating={rating} ratingAmount={ratingAmount} />
  </div>
);

Price.propTypes = {
  price: PropTypes.number.isRequired,
  priceCountry: PropTypes.string.isRequired,
  rating: PropTypes.number.isRequired,
  ratingAmount: PropTypes.number.isRequired
};

export default Price;
