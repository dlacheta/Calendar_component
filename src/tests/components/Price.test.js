import React from "react";
import { shallow } from "enzyme";
import Price from "../../components/Price";

it("Price component renders correctly", () => {
  const price = 298;
  const priceCountry = "z≈Ç";
  const rating = 4;
  const ratingAmount = 123;
  const wrapper = shallow(
    <Price
      price={price}
      priceCountry={priceCountry}
      rating={rating}
      ratingAmount={ratingAmount}
    />
  );

  expect(wrapper).toMatchSnapshot();
});
