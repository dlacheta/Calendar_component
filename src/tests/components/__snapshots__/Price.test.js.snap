// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Price component renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Price
    price={298}
    priceCountry="zł"
    rating={4}
    ratingAmount={123}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div>
          298
           
          zł
           
          <span
            className="small_text"
          >
            per night
          </span>
        </div>,
        <Rating
          rating={4}
          ratingAmount={123}
        />,
      ],
      "className": "price",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            298,
            " ",
            "zł",
            " ",
            <span
              className="small_text"
            >
              per night
            </span>,
          ],
        },
        "ref": null,
        "rendered": Array [
          298,
          " ",
          "zł",
          " ",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "per night",
              "className": "small_text",
            },
            "ref": null,
            "rendered": "per night",
            "type": "span",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "rating": 4,
          "ratingAmount": 123,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div>
            298
             
            zł
             
            <span
              className="small_text"
            >
              per night
            </span>
          </div>,
          <Rating
            rating={4}
            ratingAmount={123}
          />,
        ],
        "className": "price",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              298,
              " ",
              "zł",
              " ",
              <span
                className="small_text"
              >
                per night
              </span>,
            ],
          },
          "ref": null,
          "rendered": Array [
            298,
            " ",
            "zł",
            " ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "per night",
                "className": "small_text",
              },
              "ref": null,
              "rendered": "per night",
              "type": "span",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "rating": 4,
            "ratingAmount": 123,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
